### PR TITLE
Add version summary option to istioctl proxy-config bootstrap

### PR DESCRIPTION
**Pull request by ryantking**
https://github.com/istio/istio/issues/21517

This PR adds a `short` output type for the `istioctl proxy-config bootstrap` command that shows Istio and Envoy information in an easily human-readable format.

Example:

```
> istioctl proxy-config bootstrap -o short details-v1-79f774bdb9-6hxw6
Istio Version:       1.10.0
Istio Proxy Version: 436f365a8007cd8a13a9f1321e7cce94bcc8883e
Envoy Version:       1.18.3/Clean/RELEASE/BoringSSL
```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by ryantking** [comment]
@googlebot I signed it!

**Comment by linsun** [comment]
LGTM, thank you @ryantking.  I assume this cmd could also work?

```
istioctl proxy-config bootstrap -o short deploy/details
```

@esnible could you PTAL?

**Comment by linsun** [comment]
would be good to create a release note for this in the [release notes dir](https://github.com/istio/istio/tree/master/releasenotes).

**Comment by ryantking** [comment]
@linsun Yes, `istioctl pc bootstrap -o deploy/details-v1` will work all the same. Also added a release note, thanks for looking at it.

File: releasenotes/notes/33359.yaml
Comment (shamsher31) [inline releasenotes/notes/33359.yaml:5]:
```suggestion
  - 21517
```

File: releasenotes/notes/33359.yaml
Comment (shamsher31) [inline releasenotes/notes/33359.yaml:8]:
```suggestion
    **Added** `istioctl proxy-config bootstrap` now has a short output option (`-o short`) that shows an Istio and Envoy version summary.
```

File: istioctl/cmd/proxyconfig.go
Comment (shamsher31) [inline istioctl/cmd/proxyconfig.go:878]: Can you please also add `Example` [here](https://github.com/istio/istio/blob/d9ab7bb10dbacdc057b49321bf5b88b2128620e8/istioctl/cmd/proxyconfig.go#L834) on how to use this command. This will be helpful for docs purposes.

File: istioctl/pkg/writer/envoy/configdump/configdump.go
Comment (shamsher31) [inline istioctl/pkg/writer/envoy/configdump/configdump.go:177]: This is neat. 
image

File: istioctl/pkg/writer/envoy/configdump/testdata/configdump.json
Comment (howardjohn) [inline istioctl/pkg/writer/envoy/configdump/testdata/configdump.json:1]: could we not check in 11k lines for this test? I think we only need a very small subset

File: istioctl/cmd/proxyconfig.go
Comment (ryantking) [inline istioctl/cmd/proxyconfig.go:878]: @howardjohn @shamsher31 

The default value here is being overridden by a different command's flag setup since the commands all share a global and use `-o short` by default, which is changing the default behavior to be the new version summary versus JSON causing the istioctl pilot tests to fail.

Do we want `-o short` to be default behavior across the board for the entire command tree, if so, I'll update the tests to explicitly add a `-o json` flag. Otherwise, I'll just shadow the `outputFormat` variable in the function so it respects the json default.

Comment (linsun) [reply istioctl/cmd/proxyconfig.go:878]: my 2 cents - probably not good to change the default behavior on this, which I missed earlier while reviewing it.   There likely users building automations on this today and could break them if we change `-o short` to be the default.

Comment (ryantking) [reply istioctl/cmd/proxyconfig.go:878]: I tend to agree, the only real issue I see is that output is global to the proxy-config command so we'd be overwriting the default value based on the subcommand. Probably fine, just something worth noting.

**Comment by ryantking** [comment]
@howardjohn Can I get another set of eyes now that CI is passing and the other contents have been addressed?

**Review by howardjohn — APPROVED** [review]

**Comment by linsun** [comment]
@shamsher31 can you approve this if you are good with the latest?

**Review by linsun — APPROVED** [review]

**Review by shamsher31 — APPROVED** [review]
@ryantking  Thanks for working on this.

**Pull request merged**
